airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -361,8 +361,6 @@ def parse_response(
             response=response, stream_state=self.state, stream_slice=stream_slice, next_page_token=next_page_token
         )
         self._records_from_last_response = records
-        if records:
-            self._latest_record = records[-1]
```

Comment:
Looks like `self._latest_record` may not be used anywhere anymore, in which case I think we can remove it from `__post_init__`.